src/index.js

```diff
@@ -1,234 +1,862 @@
-          vendor: 'Payfit',
-          type: 'pay',
-          employer: companyName,
-          matchingCriterias: {
-            labelRegex: `\\b${companyName}\\b`
-          },
-          isRefund: true
-        })
-        // Extract PDF data after 06-2022
-      } else if (
-        formatedText.match(/\(Virement\)[\s,0-9,+,-]+=\s+([0-9,]+)/) &&
-        formatedText.match(/Datedepaiement\s*([0-9]{1,2}\/[0-9]{2}\/[0-9]{4})/)
-      ) {
-        const amountStg = formatedText.match(
-          /\(Virement\)[\s,0-9,+,-]+=\s+([0-9,]+)/
-        )[1]
-        const amount = parseFloat(amountStg.replace(',', '.'))
-        const dateStg = formatedText.match(
-          /Datedepaiement\s*([0-9]{1,2}\/[0-9]{2}\/[0-9]{4})/
-        )[1]
-        const date = moment(dateStg, 'DD/MM/YYYY').toDate()
-
-        Object.assign(entry, {
-          periodStart: moment(entry.date).startOf('month').format('YYYY-MM-DD'),
-          periodEnd: moment(entry.date).endOf('month').format('YYYY-MM-DD'),
-          date,
-          amount,
-          vendor: 'Payfit',
-          type: 'pay',
-          employer: companyName,
-          matchingCriterias: {
-            labelRegex: `\\b${companyName}\\b`
-          },
-          isRefund: true
+var openProxied = window.XMLHttpRequest.prototype.open
+window.XMLHttpRequest.prototype.open = function () {
+  var originalResponse = this
+  if (arguments[1] === 'https://api.payfit.com/files/files') {
+    originalResponse.addEventListener('readystatechange', function () {
+      if (originalResponse.readyState === 4) {
+        const jsonBills = JSON.parse(originalResponse.responseText)
+        bills.push(jsonBills)
+      }
+    })
+    return openProxied.apply(this, [].slice.call(arguments))
+  }
+  if (
+    typeof arguments[1] === 'string' &&
+    arguments[1].includes('/presigned-url?attachment=1')
+  ) {
+    originalResponse.addEventListener('readystatechange', function () {
+      if (originalResponse.readyState === 4) {
+        const jsonBillHref = JSON.parse(originalResponse.responseText).url
+        billsHrefs.push(jsonBillHref)
+      }
+    })
+    return openProxied.apply(this, [].slice.call(arguments))
+  } else {
+    return openProxied.apply(this, [].slice.call(arguments))
+  }
+}
+
+class PayfitContentScript extends ContentScript {
+  addSubmitButtonListener() {
+    // this.log('info', '🤖 addSubmitButtonListener')
+    const formElement = document.querySelector('form')
+    const passwordButton = document.querySelector('._button-login-password')
+    if (passwordButton) {
+      formElement.addEventListener('submit', () => {
+        const email = document.querySelector(
+          '.ulp-authenticator-selector-text'
+        )?.textContent
+        const password = document.querySelector('#password')?.value
+        this.bridge.emit('workerEvent', {
+          event: 'loginSubmit',
+          payload: { email, password }
         })
+      })
+    }
+    const error = document.querySelector('.error')
```

Comment:
I don't think this part with .error works for payfit, I took a look and other selectors are needed.